### PR TITLE
feat(Dockerfile): add health check to Dockerfile

### PR DESCRIPTION
**Pull request by nlisgo**
This commit adds a health check to the Dockerfile. The health check is set to run every 30 seconds with a timeout of 30 seconds. If the health check fails 3 times, it will mark the container as unhealthy.

### CODE REVIEW

**Comment by nlisgo** [comment]
The healthcheck should be much quicker after the passthrough feature has been implemented: https://github.com/elifesciences/enhanced-preprints-biorxiv-xslt/pull/72

File: .github/workflows/verify-api.yaml [hunk 1]
```diff
@@ -24,13 +24,20 @@ jobs:
       - name: Run api in docker container
         run: |
           docker buildx build -t epp-biorxiv-xslt:api --target api .
-          docker run -d -p 3000:80 epp-biorxiv-xslt:api
-          sleep 5 # wait for the container to start up
+          docker run -d --name api-container -p 3000:80 epp-biorxiv-xslt:api
+
+      - name: Verify that api is healthy
+        run: |
+          sleep 60 # wait for the container to start up
```
Comment (scottaubrey) [inline .github/workflows/verify-api.yaml:31]: This starting of the container consistently takes > 36s on my machine, which is quite long. GHA can be very variable in performance, and 60s maybe too short (or indeed, could be too long).

docker-compose gives a --wait command, which might be worth considering:
```sh
> cat docker-compose.yaml 
services:
  api:
    image: epp-biorxiv-xslt:api

> docker buildx build -t epp-biorxiv-xslt:api --target api . --load
[+] Building 36.1s (27/27) FINISHED                                                             docker-container:musing_swartz
# <snip build output>


>> docker-compose up --wait
[+] Running 2/2
 ✔ Network enhanced-preprints-biorxiv-xslt_default  Created                                                               0.0s 
 ✔ Container enhanced-preprints-biorxiv-xslt-api-1  Healthy                                                              36.2s 

```

**Review by scottaubrey — APPROVED** [review]
One potential improvement to pipeline times/reliability, but I'm happy as is.

File: .github/workflows/verify-api.yaml (hunk 1 above)
Comment (nlisgo) [reply .github/workflows/verify-api.yaml:31]: Sounds like a great improvement. Will add it in.

**Pull request merged**